### ISSUE TITLE
Failures during importing of gradebook data should contain better feedback

### ISSUE THREAD

**Issue by ern** [sakaiproject/sakai#3617]
We have come across a situation where a user exports data makes a few changes and then reimports the data and it fails.
![sakai___pt-d_602_01_f16___gradebook](https://cloud.githubusercontent.com/assets/209420/20566374/d838e4f4-b162-11e6-9f3f-e7d828e34f99.jpg)

In this particular case a grade item contained an asterisk in the name and when that was removed the import was fine. For the user its a matter of process of elimination to determine why the import fails.

**Comment by joliet** [comment]
Actually, the issue is that the on-screen instructions say to use the exported format, but that format fails with the error above.  The exported format uses an asterisk for comment columns, but according to Earle this character is not supported.
![gradebook_export_import_instructions](https://cloud.githubusercontent.com/assets/7387760/20566839/81d6a6e4-b164-11e6-8600-9fa68432396e.jpg)

**Comment by jonespm** [comment]
It looks like this error comes from

`				} catch (final GbImportExportInvalidFileTypeException | InvalidFormatException e) {`

So it's either if the mime detection fails or if there's a problem parsing it in poi. These seem like they could be split into two separate errors pretty easily. Do you think maybe it doesn't detect it as a csv file?

**Comment by ottenhoff** [comment]
My CSV on my Windows 10 with Chrome comes with mimetype "application/vnd.ms-excel"

Thus error:

org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportUploadStep.onSubmit org.apache.poi.openxml4j.exceptions.InvalidFormatException: Your InputStream was neither an OLE2 stream, nor an OOXML stream

**Comment by jonespm** [comment]
So I guess it would be nice to print the details of that exception to a log, but I don't see any value showing that to a user as I don't even know what it means without debugging maybe even into POI. There is a TODO in there.

`				// TODO would be nice to capture the values from these exceptions`

Stack Overflow seems to indicate that it's because of an empty file, but there is an EmptyFileException now in newer versions of POI instead of this, so most results from Stack Overflow aren't relevant?

**Comment by ottenhoff** [comment]
Problem is that code making decision based on mimetype, but if you are going to make decision on mimetype and not also file extension, then you need to be prepared for Windows with MS Office and what mimetypes will be sent. In this case, Windows is sending mimetype of Excel for a plain-text CSV file.

**Comment by jonespm** [comment]
Yeah, and it's not even "re-detecting" it (with tika), it's just trusting the mime type from the browser. That seems like it could be part of the problem and certainly led to problems in the past.

**Comment by ottenhoff** [comment]
Yes, that seems the core issue here. How is the mimetype being retrieved here. If trusting the browser, that's wrong.

**Comment by jonespm** [comment]
Yeah, it's just using Wicket (FileUpload) getcontentType. In the Wicket code it just says

`	 * The content type passed by the browser, or <code>null</code> if not defined.`

So wicket is just trusting the browser too. Arguably wicket should be using tika rather than having to pass it up to the application.

**Comment by ottenhoff** [comment]
PR #3628

**Comment by ottenhoff** [comment]
How do I indicate that this should be targeted for a merge in Sakai 11.x (11.3)?

**Comment by jeffpasch** [comment]
Hey @ottenhoff. You can just change the milestone to 11.3.
